opentype-shaping-devanagari.md

```diff
@@ -1142,13 +1142,54 @@ consonant, and all half forms.
 
 "Reph" must be moved from the beginning of the syllable to its final
 position. Because Devanagari incorporates the `REPH_POS_BEFORE_POST`
-shaping characteristic, this final position is immediately before any
-independent post-base consonant forms (meaning the first post-base
-consonant that has not formed a ligature with the base consonant).
+shaping characteristic, this final position is defined to be
+immediately before any independent post-base consonant forms (meaning
+the first post-base consonant that has not formed a ligature with the
+base consonant).
 
-  - If the syllable does not have any post-base consonants, then the
-    final "Reph" position is immediately before the first post-base
-    matra, syllable modifier, or Vedic sign.
+The algorithm for finding the final "Reph" position is
+
+<!---
+
+  - Find the first explicit "Halant" between the first post-Reph
+    consonant and the last main consonant. Move the "Reph to the
+    position immediately after this "Halant".
+	- If a zero-width joiner (ZWJ) or a zero-width non-joiner (ZWNJ)
+      follows this "Halant", move the "Reph" to the position
+      immediately after the ZWJ or ZWNJ.
+--->
+
+  - Starting at the first post-"Reph" consonant, search forward looking
+    for the first explicit "Halant", ending the search when the base
+    consonant is encountered. If such an explicit "Halant" is found,
+    move the "Reph" to the position immediately after this
+    "Halant".
+	  * If a zero-width joiner (ZWJ) or a zero-width non-joiner (ZWNJ)
+        follows this "Halant", move the "Reph" to the position
+        immediately after the ZWJ or ZWNJ. This will be the final
+        "Reph" position. 
+	  * If no ZWJ or ZWNJ follows this "Halant", leave the "Reph" in
+        its position immediately after the "Halant". This will be the
+        final "Reph" position. 
+  - If no such explicit "Halant" is found in the previous step, find
+    the first post-base consonant that has not formed a ligature with
+    the base consonant. If such a non-ligated post-base consonant is
+    found, move the "Reph" to the position immediately before the
+    non-ligated post-base consonant. This will be the final "Reph"
+    position.
+  - If no such non-ligated post-base consonant is found in the
+    previous step, move the "Reph" to the position immediately before
+    the first post-base matra, syllable modifier, or Vedic sign that
+    has a positioning tag of `POS_POSTBASE_CONSONANT` or later. This
```

Comment:
Minor: I'm fairly certain that no post-base matra, syllable modifier, or vedic sign will ever be given a `POS_*_CONSONANT` positioning tag. I understand why it was written this way, but what are your thoughts on something like:

> move the "Reph" to the position immediately after the last post-base matra, syllable modifier, or Vedic sign that has a positioning tag of `POS_BEFORE_POST` or later.

Reply:
I think your assessment is correct. Looking at this again, though, I also think that just dictating the positioning tag on an each-script basis is potentially more confusing than it ought to be. 

So, I'm thinking we could put:
> move the "Reph" to the position immediately after the last post-base matra, syllable modifier, or Vedic sign that has a positioning tag after the script's Reph position.

...and it would be identical for each script, which is a plus and communicates the intent. Then I can slip in a note clarifying what that position is and what to expect.

Reply:
Nice - that seems very sensible!